satellite/satellitedb/orders.go

```diff
@@ -87,7 +98,107 @@ func (db *ordersDB) SettleRemoteOrder(ctx context.Context, orderLimit *pb.OrderL
+	bucketBandwidth, err := tx.Find_BucketBandwidthRollup_By_BucketId_And_IntervalStart_And_Action(
+		ctx,
+		dbx.BucketBandwidthRollup_BucketId(dbxSerialNumber.BucketId),
+		dbx.BucketBandwidthRollup_IntervalStart(intervalStart),
+		dbx.BucketBandwidthRollup_Action(uint(orderLimit.Action)),
+	)
+	if err != nil {
+		return err
+	}
+	if bucketBandwidth == nil {
+		_, err = tx.Create_BucketBandwidthRollup(
+			ctx,
+			dbx.BucketBandwidthRollup_BucketId(dbxSerialNumber.BucketId),
+			dbx.BucketBandwidthRollup_IntervalStart(intervalStart),
+			dbx.BucketBandwidthRollup_IntervalSeconds(0),
+			dbx.BucketBandwidthRollup_Action(uint(orderLimit.Action)),
+			dbx.BucketBandwidthRollup_Inline(0),
+			dbx.BucketBandwidthRollup_Allocated(0),
+			dbx.BucketBandwidthRollup_Settled(uint64(order.Amount)),
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		update := dbx.BucketBandwidthRollup_Update_Fields{
+			Settled: dbx.BucketBandwidthRollup_Settled(bucketBandwidth.Settled + uint64(order.Amount)),
+		}
+		_, err = tx.Update_BucketBandwidthRollup_By_BucketId_And_IntervalStart_And_Action(ctx,
+			dbx.BucketBandwidthRollup_BucketId(bucketBandwidth.BucketId),
+			dbx.BucketBandwidthRollup_IntervalStart(bucketBandwidth.IntervalStart),
+			dbx.BucketBandwidthRollup_Action(bucketBandwidth.Action),
+			update,
+		)
+		if err != nil {
+			return err
+		}
+	}
+
+	nodeBandwidth, err := tx.Find_StoragenodeBandwidthRollup_By_StoragenodeId_And_IntervalStart_And_Action(
+		ctx,
+		dbx.StoragenodeBandwidthRollup_StoragenodeId(orderLimit.StorageNodeId.Bytes()),
+		dbx.StoragenodeBandwidthRollup_IntervalStart(intervalStart),
+		dbx.StoragenodeBandwidthRollup_Action(uint(orderLimit.Action)),
+	)
+	if err != nil {
+		return err
+	}
+
+	if nodeBandwidth == nil {
+		_, err = tx.Create_StoragenodeBandwidthRollup(
+			ctx,
+			dbx.StoragenodeBandwidthRollup_StoragenodeId(orderLimit.StorageNodeId.Bytes()),
+			dbx.StoragenodeBandwidthRollup_IntervalStart(intervalStart),
+			dbx.StoragenodeBandwidthRollup_IntervalSeconds(0),
+			dbx.StoragenodeBandwidthRollup_Action(uint(orderLimit.Action)),
+			dbx.StoragenodeBandwidthRollup_Allocated(0),
+			dbx.StoragenodeBandwidthRollup_Settled(uint64(order.Amount)),
+		)
+		if err != nil {
+			return err
+		}
+	} else {
+		update := dbx.StoragenodeBandwidthRollup_Update_Fields{
+			Settled: dbx.StoragenodeBandwidthRollup_Settled(bucketBandwidth.Settled + uint64(order.Amount)),
+		}
+		_, err = tx.Update_StoragenodeBandwidthRollup_By_StoragenodeId_And_IntervalStart_And_Action(
+			ctx,
+			dbx.StoragenodeBandwidthRollup_StoragenodeId(nodeBandwidth.StoragenodeId),
+			dbx.StoragenodeBandwidthRollup_IntervalStart(nodeBandwidth.IntervalStart),
+			dbx.StoragenodeBandwidthRollup_Action(uint(nodeBandwidth.Settled)),
+			update,
+		)
+		if err != nil {
+			return err
+		}
+	}
 
 	return nil
 }
+
+// GetBucketBandwidth
+func (db *ordersDB) GetBucketBandwidth(ctx context.Context, bucketID []byte, from, to time.Time) (int64, error) {
+	var sum *int64
```

Comment:
does this need to be a pointer? could it be `var sum int64`?

Reply:
Unfortunately sqlite will return NULL if there are no rows, so using `var sum int64` would fail when there are no rows.

Reply:
oh interesting.   There are many places in the code base that use a value for the arg to `.Scan()`.  Are those incorrect then? 

Here is one example:
https://github.com/storj/storj/blob/master/pkg/piecestore/psserver/psdb/psdb.go#L401

Reply:
In case of selecting a single row (not using SUM), we get no rows... so that should be fine and the err is properly `sql.ErrNoRows`